pkg_resources/__init__.py

```diff
@@ -2722,11 +2736,34 @@ def requires(self, extras=()):
                 )
         return deps
 
+    def _get_metadata_path_for_display(self, name):
+        """
+        Return the path to the given metadata file, if available.
+        """
+        try:
+            # We need to access _get_metadata_path() on the provider object
+            # directly rather than through this class's __getattr__()
+            # since _get_metadata_path() is marked private.
+            path = self._provider._get_metadata_path(name)
+
+        # Handle exceptions e.g. in case the distribution's metadata
+        # provider doesn't support _get_metadata_path().
+        except Exception:
+            return '[could not detect]'
```

Comment:
The coverage is showing this as hit, but I don't see a test that actually hits it, do you know why that would be? Unexplained coverage worries me a bit.

Reply:
It looks like there were at least a few tests hitting that line. Here is the simplest example of those tests (`pkg_resources/tests/test_resources.py::TestDistro::testDistroDependsOptions`): https://github.com/pypa/setuptools/blob/16e452a42a3dbbb0ab3d3146ffa3b743cdca2539/pkg_resources/tests/test_resources.py#L326

Specifically, this line triggered it: https://github.com/pypa/setuptools/blob/16e452a42a3dbbb0ab3d3146ffa3b743cdca2539/pkg_resources/tests/test_resources.py#L349-L350

The test "manually" constructs a `Distribution` object here: https://github.com/pypa/setuptools/blob/16e452a42a3dbbb0ab3d3146ffa3b743cdca2539/pkg_resources/tests/test_resources.py#L138-L139

I updated the PR with another test case to explicitly test this code path. The test case does something similar to the example above (constructing a `Distribution` object manually). I also addressed your other review comment in the updated PR.